book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd

```diff
@@ -124,49 +124,26 @@ The `batch_run.R` script can take the following command-line arguments:
 
 Each build starts by launching a separate clean virtual machine for each R version and performs roughly the following actions on all of them:
-
-* Installs binary system dependencies needed by PEcAn (NetCDF and HDF5 libraries, JAGS, udunits, etc).
-* Installs all the R packages that are declared as dependencies in any PEcAn package, as computed by `scripts/generate_dependencies.R`.
-* Clones the PEcAn repository from GitHub, and checks out the branch to be tested.
-* Retrieves any cached files available from previous Travis builds.
-    - The main thing in the cache is previously-installed dependency R packages, to avoid recompiling them every time.
-    - If the cache becomes stale or is preventing a package update needed by the build (e.g. to get a new version that contains a needed bug fix), delete the cache through the Travis web interface and it will be reconstructed on the next build.
-    - Because PEcAn has so many dependencies, builds with no cache will spend most of their time recompiling packages and will probably run out of time before the tests complete. You can fix this by using `scripts/travis/cache_buildup.sh` and `scripts/travis/prime_travis_cache.sh` to build up the cache incrementally through one or more [custom builds](https://blog.travis-ci.com/2017-08-24-trigger-custom-build), each of which installs some dependencies and then uploads a freshened cache *without* running any tests. Once all dependencies have been cached, restart the standard full build.
-* Initializes a skeleton version of the PEcAn database (BeTY) containing a few public records to be used by the test runs.
-* Installs all the PEcAn packages, recompiling documentation and installing dependencies as needed.
-* Runs package unit tests (the same ones you run locally with `make test` or `devtools::test(pkgname)`).
-    - As discussed in [Unit testing](#developer-testing-unit), these tests should run quickly and test individual components in relative isolation.
-    - Any test that calls the `skip_on_ci` function will be skipped. This is useful for tests that need to run for a very long time (e.g. large data product downloads) or require resources that aren't available on Travis (e.g. specific models), but be sure to run these tests locally before pushing your code!
-* Runs R package checks (the same ones you run locally with `make check` or `devtools::check(pkgname)`), skipping tests and documentation rebuild because we just did those in the previous steps.
-    - Any ERROR in the check output will stop the build immediately.
-    - If there are no ERRORs, any WARNINGs or NOTEs are compared against a stored historic check result in `<package>/tests/Rcheck_reference.log`. If the package has no stored reference result, all WARNINGs and NOTEs are considered newly added and reported as build failures.
```

Comment:
I believe this section is the only written documentation of how the cached check comparisons work. Should it be retained (possibly in another location)?

Reply:
use to big of an axe.

Reply:
added text back